mlflow/utils/model_utils.py

```diff
@@ -285,9 +316,16 @@ def _add_code_from_conf_to_system_path(local_path, conf, code_key=FLAVOR_CONFIG_
     """
     assert isinstance(conf, dict), "`conf` argument must be a dict."
 
-    if code_key in conf and conf[code_key]:
-        code_path = os.path.join(local_path, conf[code_key])
+    if code_value := conf.get(code_key):
+        code_path = os.path.join(local_path, code_value)
         _add_code_to_system_path(code_path)
+        start_modules = set(sys.modules)
+        try:
+            yield
+        finally:
+            _restore_sys_path_and_modules({code_path}, start_modules)
```

Comment:
this seems to be problemetic:

it  does :

```
with _add_code_from_conf_to_system_path(..):
   model_impl = load_pyfuc(...)
```
but then we need to use `model_impl` in follow-up code, but `_restore_sys_path_and_modules` removes the dependent modules.

Reply:
I see, can you create an example where the current implementation doesn't work

Reply:
Why is the test passing? What is missing?

Reply:
You can do "importing any of the custom code path module after `mlflow.pyfunc.load_model`", then it will fail.

> Why is the test passing

Passed test only uses the loaded model. the loaded model has the reference to the custom module object even it is cleaned from `sys.path` or `sys.modules`

Reply:
Good point. I think preserving that behavior with this PR is challenging.

Reply:
Documenting this behavior and providing a workaround is another option.

Reply:
Another option is only restore `sys.modules`.